src/emit/substrate/target.rs

```diff
@@ -361,14 +361,20 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
 
         binary.builder.position_at_end(retrieve_block);
 
-        let offset = unsafe {
-            binary.builder.build_gep(
-                binary.context.i8_type().array_type(SCRATCH_SIZE),
-                binary.scratch.unwrap().as_pointer_value(),
-                &[i32_zero!(), index],
-                "data_offset",
+        fn function_to_get_pointer(binary: &Binary, index: &IntValue)-> PointerValue{
```

Comment:
This name is not good. The operation here indexes the scratch array, not only returns a pointer to something. The name should describe what this function does.

Reply:
got it I will surely try to work upon this, thank you so much for your input.